homeassistant/components/proxmoxve/__init__.py

```diff
@@ -188,67 +188,61 @@ def create_coordinator_container_vm(
     host_name: str,
     node_name: str,
     vm_id: int,
-    vm_type: int,
+    api_category: ProxmoxType,
 ) -> DataUpdateCoordinator[dict[str, Any] | None]:
     """Create and return a DataUpdateCoordinator for a vm/container."""
 
-    async def async_update_data() -> dict[str, Any] | None:
+    async def async_update_data() -> dict[str, ProxmoxVMData] | None:
         """Call the api and handle the response."""
 
         def poll_api() -> dict[str, Any] | None:
             """Call the api."""
-            vm_status = call_api_container_vm(proxmox, node_name, vm_id, vm_type)
-            return vm_status
+            api_status = {}
 
-        vm_status = await hass.async_add_executor_job(poll_api)
-
-        if vm_status is None:
-            _LOGGER.warning(
+            try:
+                if api_category == ProxmoxType.QEMU:
+                    api_status = (
+                        proxmox.nodes(node_name).qemu(vm_id).status.current.get()
+                    )
+                elif api_category == ProxmoxType.LXC:
+                    api_status = (
+                        proxmox.nodes(node_name).lxc(vm_id).status.current.get()
+                    )
+            except (
+                AuthenticationError,
+                SSLError,
+                ConnectTimeout,
+                ResourceException,
+            ) as error:
+                raise UpdateFailed from error
+
+            LOGGER.debug("API Response: %s", api_status)
+            return api_status
+
+        api_status = await hass.async_add_executor_job(poll_api)
+
+        if api_status is None:
+            LOGGER.warning(
                 "Vm/Container %s unable to be found in node %s", vm_id, node_name
             )
             return None
 
-        return parse_api_container_vm(vm_status)
+        api_result = {}
+        api_result["result"] = ProxmoxVMData(
+            status=api_status["status"],
+            name=api_status["name"],
+        )
+        return api_result
```

Comment:
I was only able to solve the type hints like this... I don't know if it's a good way.